context.go

```diff
@@ -313,9 +313,19 @@ func checkRequiredFlags(flags []Flag, context *Context) requiredFlagsErr {
 	var missingFlags []string
 	for _, f := range flags {
 		if rf, ok := f.(RequiredFlag); ok && rf.IsRequired() {
-			key := strings.Split(f.GetName(), ",")[0]
-			if !context.IsSet(key) {
-				missingFlags = append(missingFlags, key)
+			key := strings.Split(f.GetName(), ",")
+			shortName := strings.TrimSpace(key[0])
+			if len(key) > 1 {
+				// has short name
+				longName := strings.TrimSpace(key[1])
+				if !context.IsSet(shortName) && !context.IsSet(longName) {
+					missingFlags = append(missingFlags, key[0])
+				}
+			} else {
+				// does not have short name
+				if !context.IsSet(shortName) {
+					missingFlags = append(missingFlags, key[0])
+				}
```

Comment:
Is it mandated the order in which short vs non short should be defined? Also the comments in the if branches seem the other way around judging by the if condition

Reply:
The names here are only for readability. Even if you define it the other way, the code will still work. Essentially what we are trying to do here is to check if both the defined values are there in the context and only if both of them are absent, throw the error

Reply:
In case of definition "n, names" shortName = n and longName = names
In case of definition "names, n" shortName = names and longName = n

Either  way the code would evaluate the same :)

Reply:
Sure but its incredibly confusing for the reader

Reply:
would you prefer just key[0] and key[1] which would be even more ambiguous?

Reply:
I have changed the names of the variables now. Please check